File: Assets/InsightARWorld/InsightARExporter/Utility/BuiltinLuaScripts/Landmark/LandmarkResource.lua


local LandmarkResource = {}


function LandmarkResource:New(game_object)
    if self ~= LandmarkResource then
        return nil, "LandmarkResource:New is nil"
    end
    local newinstance = setmetatable({}, {__metatable = {}, __index = LandmarkResource})
    newinstance.transform = game_object.transform
    transform = game_object.transform
    newinstance.gameObject = game_object
    gameObject = game_object
    return newinstance
end

function LandmarkResource:Awake()
	
	
end

function LandmarkResource:Start()
	
end

-- init
function LandmarkResource:Init()
	self.canvasTrans = Insight.GameObject.Find("HintController").transform;
	self.mainCamera =  Insight.GameObject.Find("Main Camera"):GetComponent("Camera");
	self.mainCameraTrans = self.mainCamera.transform;
	
	-- init panel
	self.initPanelTrans = self.canvasTrans:Find("initpanel");
	self.initIconTrans = self.initPanelTrans:Find("initicon");
	self.initSlideTrans = self.initPanelTrans:Find("initslide");
	self.initSlideImage = self.initSlideTrans.gameObject:GetComponent("Image");
	self.initSlideMaterial = self.initSlideImage.material;
	self.initOKTrans = self.initPanelTrans:Find("initok");
	
	-- model
	self.modelRootTrans = Insight.GameObject.Find("World").transform;
	-- self.jinyujiTrans = self.modelRootTrans:Find("JINYUJI");
	-- self.datiangouTrans = self.modelRootTrans:Find("DATIANGOU");
	-- self.yijinzhentianTrans = self.modelRootTrans:Find("YIJINZHENTIAN");
	-- self.jinlongTrans = self.modelRootTrans:Find("jinlong");
	-- self.balloonTrans = self.modelRootTrans:Find("fire_balloons");
	-- self.landmarkTrans = self.modelRootTrans:Find("LandMark_C11");
	-- self.axisTrans = self.modelRootTrans:Find("Axis");
	-- self.maskTrans = self.modelRootTrans:Find("C11_Mask");
	
	--trackpanel
	self.trackPanelTrans = self.canvasTrans:Find("trackpanel");
	self.upIconTrans = self.trackPanelTrans:Find("upicon");
	self.downIconTrans = self.trackPanelTrans:Find("downicon");
	self.trackLimitedTrans = self.trackPanelTrans:Find("tracklimitedicon");
end

return LandmarkResource

File: U3DExporter/2022_10_31_10_52_29_Android/Assets/InsightARWorld/InsightARExporter/Utility/BuiltinJSScripts/Landmark/LandmarkResource.js
function LandmarkResource(gameObject)
{
    this.transform = gameObject.transform;
    this.gameObject = gameObject;
}
//Write prototype function here
LandmarkResource.prototype = Object.assign(Object.create(Object.prototype), {
    // Start is called before the first frame update
    Start: function(){
        
    },
    Awake: function(){

    },
    // Update is called once per frame
    Update: function()
    {
        
    },
    Init: function()
    {
        this.canvasTrans = Insight.GameObject.Find("HintController").transform;
        this.mainCamera =  Insight.GameObject.Find("Main Camera").getComponent("Camera");
        this.mainCameraTrans = this.mainCamera.transform;
        
        // init panel
        this.initPanelTrans = this.canvasTrans.find("initpanel");
        this.initIconTrans = this.initPanelTrans.find("initicon");
        this.initSlideTrans = this.initPanelTrans.find("initslide");
        this.initSlideImage = this.initSlideTrans.gameObject.getComponent("Image");
        this.initSlideMaterial = this.initSlideImage.material;
        this.initOKTrans = this.initPanelTrans.find("initok");
        
        // model
        this.modelRootTrans = Insight.GameObject.Find("World").transform;
        
        //trackpanel
        this.trackPanelTrans = this.canvasTrans.find("trackpanel");
        this.upIconTrans = this.trackPanelTrans.find("upicon");
        this.downIconTrans = this.trackPanelTrans.find("downicon");
        this.trackLimitedTrans = this.trackPanelTrans.find("tracklimitedicon");
        this.trackFailTrans = this.trackPanelTrans.find("trackfailicon");
        this.trackFailButton = this.trackFailTrans.find("trackfailbutton").gameObject.getComponent("Button");
        this.errorCodeTrans = this.trackPanelTrans.find("errorcodeicon");
        this.errorCodeButton = this.errorCodeTrans.find("errorcodebutton").gameObject.getComponent("Button");
        this.errorCodeText = this.errorCodeTrans.find("errorcodetext").gameObject.getComponent("Text");
    }
});

//Return the script module
LandmarkResource